pkg/kubeadm/cmd/init.go

```diff
@@ -59,11 +60,11 @@ func NewCmdInit(out io.Writer, params *kubeadmapi.BootstrapParams) *cobra.Comman
 
 func RunInit(out io.Writer, cmd *cobra.Command, args []string, params *kubeadmapi.BootstrapParams) error {
 	if params.Discovery.ListenIP == "" {
-		ip, err := kubeadmutil.GetDefaultHostIP()
+		ip, err := netutil.ChooseHostInterface()
 		if err != nil {
 			return err
 		}
-		params.Discovery.ListenIP = ip
+		params.Discovery.ListenIP = string(ip)
```

Comment:
Yes, it probably should be, but the whole structure needs reviewing, and I
would leave that for later.

On Thu, 1 Sep 2016, 11:29 Lucas Käldström, <notifications@github.com> wrote:

> In pkg/kubeadm/cmd/init.go
> <https://github.com/kubernetes/kubernetes/pull/31221#discussion_r77149880>
> :
>
> >  		if err != nil {
> >  			return err
> >  		}
> > -		params.Discovery.ListenIP = ip
> > +		params.Discovery.ListenIP = string(ip)
>
> Why not have ListenIP of type net.IP?
>
> —
> You are receiving this because you were mentioned.
>
>
> Reply to this email directly, view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/31221/files/57096ca902c5c666d45df191037d9278972af1fd..be3fdcbc3810867019c660598be3e8b7e8f99973#r77149880>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAPWS1_fp-mnMUtxiVf_cTtLVZP2W_Mlks5qlqkNgaJpZM4JqsTH>
> .
>